src/pages/ServiceGraph/SummaryPanelGroup.tsx

```diff
@@ -49,37 +51,45 @@ export default class SummaryPanelGroup extends React.Component<SummaryPanelPropT
   componentDidMount() {
     this._isMounted = true;
     this.updateRpsCharts(this.props);
+    this.updateHealth();
   }
 
   componentDidUpdate(prevProps: SummaryPanelPropType) {
     if (shouldRefreshData(prevProps, this.props)) {
       this.updateRpsCharts(this.props);
+      this.updateHealth();
     }
   }
 
   componentWillUnmount() {
     this._isMounted = false;
   }
 
+  updateHealth() {
+    const healthPromise = this.props.data.summaryTarget.data('healthPromise');
+    if (healthPromise) {
+      healthPromise.then(h => this.setState({ health: h })).catch(err => this.setState({ health: undefined }));
+    }
```

Comment:
Perhaps you should do `this.setState({ health: null }))` if there is no `healthPromise`.